### PR TITLE
Feature/confirm email

### PR DESCRIPTION
**Pull request by h4570**
### Working with [Product Backlog #11 - Registration email confirmation](https://dev.azure.com/tasob/floiir/_sprints/taskboard/floiir%20Team/floiir/Sprint%201?workitem=11)

**Features:**
- Email builder
- 1 template of email (BlueGray)
- WebApi i18n factory
- Business logic folder structure refactor
- Register endpoint refactor
- Confirm email component

### CODE REVIEW

File: backend/WebApi/Resources/EmailTemplates/BlueGray/beginning.html [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        <!-- LOGO -->
+        <tr>
+            <td bgcolor="{{bgColor}}" align="center">
+                <table border="0" cellpadding="0" cellspacing="0" width="100%" style="max-width: 600px;">
+                    <tr>
+                        <td align="center" valign="top" style="padding: 40px 10px 40px 10px;"> </td>
+                    </tr>
+                </table>
+            </td>
+        </tr>
```
Comment (Avertere7) [inline backend/WebApi/Resources/EmailTemplates/BlueGray/beginning.html:117]: new line :) #lint

File: backend/WebApi/Resources/EmailTemplates/BlueGray/button.html [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+                                            </tr>
+                                        </table>
+                                    </td>
+                                </tr>
+                            </table>
+                        </td>
+                    </tr>
+                </table>
+            </td>
+        </tr>
```
Comment (Avertere7) [inline backend/WebApi/Resources/EmailTemplates/BlueGray/button.html:21]: new line

File: backend/WebApi/Resources/EmailTemplates/BlueGray/centered-img.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+            <td bgcolor="{{bgColor}}" align="center" style="padding: 0px 10px 0px 10px;">
+                <table border="0" cellpadding="0" cellspacing="0" width="100%" style="max-width: 600px;">
+                    <tr>
+                        <td bgcolor="#ffffff" align="center" valign="top" style="padding: 10px 20px 0px 20px; border-radius: 4px 4px 0px 0px; color: #111111; font-family: 'Lato', Helvetica, Arial, sans-serif; font-size: 48px; font-weight: 400; letter-spacing: 4px; line-height: 48px;">
+                            <img src="{{url}}" width="125" height="120" style="display: block; border: 0px;" />
+                        </td>
+                    </tr>
+                </table>
+            </td>
+        </tr>
```
Comment (Avertere7) [inline backend/WebApi/Resources/EmailTemplates/BlueGray/centered-img.html:11]: new line

File: backend/WebApi/Resources/EmailTemplates/BlueGray/ending.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+                            <p style="margin: 0;">Floiir - smart space management system.</p>
+                        </td>
+                    </tr>
+                </table>
+            </td>
+        </tr>
+    </table>
+</body>
+
+</html>
```
Comment (Avertere7) [inline backend/WebApi/Resources/EmailTemplates/BlueGray/ending.html:15]: new line

File: backend/WebApi/Resources/EmailTemplates/BlueGray/header.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+            <td bgcolor="{{bgColor}}" align="center" style="padding: 0px 10px 0px 10px;">
+                <table border="0" cellpadding="0" cellspacing="0" width="100%" style="max-width: 600px;">
+                    <tr>
+                        <td bgcolor="#ffffff" align="center" valign="top" style="padding: 10px 20px 0px 20px; border-radius: 4px 4px 0px 0px; color: #111111; font-family: 'Lato', Helvetica, Arial, sans-serif; font-size: 48px; font-weight: 400; letter-spacing: 4px; line-height: 48px;">
+                            <h1 style="font-size: 48px; font-weight: 400; margin: 2;">{{text}}</h1> 
+                        </td>
+                    </tr>
+                </table>
+            </td>
+        </tr>
```
Comment (Avertere7) [inline backend/WebApi/Resources/EmailTemplates/BlueGray/header.html:11]: new line

File: backend/WebApi/Resources/EmailTemplates/BlueGray/spacer.html [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+            <td bgcolor="{{bgColor}}" align="center" style="padding: 0px 10px 0px 10px;">
+                <table border="0" cellpadding="0" cellspacing="0" width="100%" style="max-width: 600px;">
+                    <tr>
+                        <td bgcolor="#ffffff" align="center" valign="top" style="padding: 40px 20px 0px 20px; border-radius: 4px 4px 0px 0px; color: #111111; font-family: 'Lato', Helvetica, Arial, sans-serif; font-size: 48px; font-weight: 400; letter-spacing: 4px; line-height: 48px;">
+                            <!-- SPACER -->
+                        </td>
+                    </tr>
+                </table>
+            </td>
+        </tr>
```
Comment (Avertere7) [inline backend/WebApi/Resources/EmailTemplates/BlueGray/spacer.html:11]: new line

File: backend/WebApi/Resources/EmailTemplates/BlueGray/text.html [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+            <td bgcolor="{{bgColor}}" align="center" style="padding: 0px 10px 0px 10px;">
+                <table border="0" cellpadding="0" cellspacing="0" width="100%" style="max-width: 600px;">
+                    <tr>
+                        <td bgcolor="#ffffff" align="left" style="padding: 20px 30px 0px 30px; color: #666666; font-family: 'Lato', Helvetica, Arial, sans-serif; font-size: 18px; font-weight: 400; line-height: 25px;">
+                            <p style="margin: 0;">{{text}}</p>
+                        </td>
+                    </tr>
+                </table>
+            </td>
+        </tr>
```
Comment (Avertere7) [inline backend/WebApi/Resources/EmailTemplates/BlueGray/text.html:11]: new line

File: frontend/web-app/src/app/components/pages/confirm-email/confirm-email.component.html [hunk 8]
```diff
@@ -1 +1,22 @@
...
+                It can take up to 10 minutes to receive this email.
+            </div>
+
+            <div class="confirm-email-resend">
+                Didn't receive the email? <span class="send-again-text" (click)="onResendEmailClick()">Send it
+                    again</span>
+            </div>
+        </div>
+    </div>
+</div>
```
Comment (Avertere7) [inline frontend/web-app/src/app/components/pages/confirm-email/confirm-email.component.html:22]: new line

File: frontend/web-app/src/app/components/pages/confirm-email/confirm-email.component.scss [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+        color: blue;
+        text-decoration: underline;
+        cursor: pointer;
+    }
+
+}
+
+.material-icons {
+    vertical-align: middle !important;
+}
```
Comment (Avertere7) [inline frontend/web-app/src/app/components/pages/confirm-email/confirm-email.component.scss:31]: new line

Comment (h4570) [reply frontend/web-app/src/app/components/pages/confirm-email/confirm-email.component.scss:31]: Can do it, but not sure if should. VSCode by default is removing eol in scss

**Pull request merged**